cms/src/test/java/uk/nhs/digital/externalstorage/workflow/externalFilePublish/ExternalFilePublishTaskTest.java

```diff
@@ -30,33 +41,163 @@
 
     private ExternalFilePublishTask externalFilePublishTask;
     private Session session;
-
     private Node rootNode;
+    private Node testFolder;
+    private final String testExternalReference = "ab/1c2d3e/one.pdf";
 
     @Before
     public void setUp() throws RepositoryException {
         initMocks(this);
         given(s3Connector.publishResource(any(String.class)))
             .willReturn(true);
+        given(s3Connector.unpublishResource(any(String.class)))
+            .willReturn(true);
 
         Repository repository = MockJcr.newRepository();
         session = repository.login();
         rootNode = session.getRootNode();
+        testFolder = rootNode.addNode("lorem-ipsum", NT_FOLDER);
         externalFilePublishTask = new ExternalFilePublishTask();
 
         given(workflowContext.getInternalWorkflowSession()).willReturn(session);
         externalFilePublishTask.setWorkflowContext(workflowContext);
     }
 
     @Test
-    public void shouldCallS3() throws WorkflowException, RepositoryException {
-        Node docNode = rootNode.addNode("test-document");
-        addVariantNode(docNode, UNPUBLISHED);
-        MockJcr.setQueryResult(session, getQueryResults());
+    public void shouldPublishResources() throws WorkflowException, RepositoryException {
+        MockJcr.addQueryResultHandler(session, searchResourcesHandle(testExternalReference));
+
+        executeTask(getDocumentHandleWithVariant(getRandomDocumentType(), UNPUBLISHED), externalFilePublishTask);
+
+        then(s3Connector).should().publishResource(testExternalReference);
+        then(s3Connector).should(never()).unpublishResource(anyString());
+    }
+
+    @Test
+    public void shouldPublishFinalisedPublicationResources() throws WorkflowException, RepositoryException {
```

Comment:
Isn't this test made redundant by `shouldPublishFinalisedPublicationAndItsDatasetsResources`?

Reply:
renamed to `shouldPublishFinalisedStandalonePublicationResources`